e2e/tests/openmrs-superset-flows.spec.ts

```diff
@@ -419,9 +441,179 @@ test('Voiding an OpenMRS obs updates the observation in Superset observations ta
 
   await page.getByRole('tab', { name: 'Results' }).click();
   await superset.clearSQLEditor();
+  await openmrs.deletePatient();
 });
 
-test.afterEach(async ({ page }) => {
+test('Voiding an OpenMRS patient updates the patient in Superset patients table.', async ({ page }) => {
+  // setup
+  await openmrs.createPatient();
+  await openmrs.searchPatientId();
+  const patientIdentifier = await page.locator('#demographics section p:nth-child(2)').textContent();
+  await superset.open();
+  await expect(page).toHaveURL(/.*superset/);
+  await superset.selectDBSchema();
+  await superset.clearSQLEditor();
+  let personVoidedQuery = `SELECT person_voided FROM patients where identifiers like 'OpenMRS ID: ${patientIdentifier}';`;
+  await page.getByRole('textbox').first().fill(personVoidedQuery);
+  await superset.runSQLQuery();
+  let patientVoidedState = await page.locator('div.virtual-table-cell');
+
+  await expect(patientVoidedState).toContainText('false');
+
+  // replay
+  await openmrs.deletePatient();
```

Comment:
This particular test case invokes `openmrs.deletePatient()` during the replay thus voiding the patient. This implies that the method can not be called again in the cleanup function (afterEach()) with this particular test so the suggested approach is to let each test case invoke the function that voids  the patient.

Reply:
If deletions are actually voids in OpenMRS, I'd rather change `openmrs.deletePatient()` to `openmrs.voidPatient()`.